File: SmartLink/Source/SmartLinkClient/NRC_ServiceGenericClasses/TaskSchedulerClasses/TaskScheduler_DailyScheduler.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
//********************************************************************'
// Created by Elibad - 09/01/2007
//
//  This class inheriths the TaskScheduler functionality
//  and implements the Daily Schedule
//*********************************************************************
namespace NRC.Miscellaneous.TaskScheduler
{
	public class DailyScheduler : Scheduler
	{


		protected override DateTime GetNextOccurrence(DateTime BaseDate)
		{
			DateTime functionReturnValue = default(DateTime);
			if (System.DateTime.Now.ToString("HH:mm:ss").CompareTo(this.Time) > 0) {
				functionReturnValue = System.DateTime.Parse(BaseDate.AddDays(this.Interval).ToString("MM/dd/yyyy") + " " + this.Time);
			} else {
				functionReturnValue = System.DateTime.Parse(BaseDate.ToString("MM/dd/yyyy") + " " + this.Time);
			}
			this._bFirstRun = false;
			return functionReturnValue;
		}

		public DailyScheduler() : base()
		{
		}

		public DailyScheduler(string Time) : base(Time)
		{
		}

		public DailyScheduler(int Interval, string StartDate, string Time, bool Enabled) : base(Interval, StartDate, Time, Enabled)
		{
		}

	}




}

File: SmartLink/Source/SmartLinkClient/NRC_ServiceGenericClasses/TaskSchedulerClasses/TaskScheduler_DailyScheduler.vb
Option Explicit On
'********************************************************************'
' Created by Elibad - 09/01/2007
'
'  This class inheriths the TaskScheduler functionality
'  and implements the Daily Schedule
'*********************************************************************
Namespace Miscellaneous.TaskScheduler
    Public Class DailyScheduler : Inherits Scheduler


        Protected Overrides Function GetNextOccurrence(ByVal BaseDate As DateTime) As DateTime
            If Date.Now.ToString("HH:mm:ss") > Me.Time Then
                GetNextOccurrence = Date.Parse(BaseDate.AddDays(Me.Interval).ToString("MM/dd/yyyy") & " " & Me.Time)
            Else
                GetNextOccurrence = Date.Parse(BaseDate.ToString("MM/dd/yyyy") & " " & Me.Time)
            End If
            Me._bFirstRun = False
        End Function

        Public Sub New()
            MyBase.New()
        End Sub

        Public Sub New(ByVal Time As String)
            MyBase.New(Time)
        End Sub

        Public Sub New(ByVal Interval As Integer, ByVal StartDate As String, ByVal Time As String, ByVal Enabled As Boolean)
            MyBase.New(Interval, StartDate, Time, Enabled)
        End Sub

    End Class




End Namespace

